plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/geoip/GeoIpProcessor.java

```diff
@@ -68,13 +71,21 @@ public void execute(Data data) {
             throw new RuntimeException(e);
         }
 
-        final Map<String, Object> geoData;
+        Map<String, Object> geoData;
         switch (dbReader.getMetadata().getDatabaseType()) {
             case "GeoLite2-City":
-                geoData = retrieveCityGeoData(ipAddress);
+                try {
+                    geoData = retrieveCityGeoData(ipAddress);
+                } catch (Exception e) {
+                    geoData = Collections.singletonMap("error", ExceptionsHelper.detailedMessage(e));
```

Comment:
Ok, the way I see this is that this isn't a failure, because the the geoip processor has simply no extra data to enrich the document with. I've updated the PR to make this clearer.

I agree with general failures.